plugins/logs/plugin.go

```diff
@@ -650,9 +664,17 @@ func (p *Plugin) reconfigure(config interface{}) {
 }
 
 func (p *Plugin) bufferChunk(buffer *logBuffer, bs []byte) {
-	dropped := buffer.Push(bs)
-	if dropped > 0 {
-		p.logger.Error("Dropped %v chunks from buffer. Reduce reporting interval or increase buffer size.", dropped)
+	if p.limiter != nil {
```

Comment:
I was surprised to see the DPS limit enforced here and not before we write the event into the encoder, but now I see that we're enforcing the buffer size limit here as well. I'm just wondering if we need to do something else because the chunk could contain >1 decision...

Reply:
We could enforce the rate limit before we encode the event. This means that when a chunk upload fails, we'll have to decode the chunk to get the decision events and encode the event again based on the rate limit logic. WDYT ?

Reply:
Updated the code to enforce the rate limit before encoding the event. LMKWYT.

Reply:
Looks fine to me. I hadn't thought too much about the failure scenario, but this makes sense. If the upload is failing, we don't want to buffer events until the issue resolves itself as that could result in a storm of events being uploaded.